src/test/java/org/example/BaseTests.java

```diff
@@ -88,6 +109,18 @@ public void recordFailure(ITestResult result) {
 			} catch (IOException e) {
 				e.printStackTrace();
 			}
+			jiraService.createIssue("Test failed: " + result.getName());
+		}
+	}
+
+	@AfterMethod
```

Comment:
AfterMethod is not help to receive the notification after the all tests are done. Even with this implementation it will send the slack message after every test, I guess that it will be better to move this implementation to the AfterClass at least (and provide the link to the last test run result or/and the short statistic how many tests are passed/failed and/or skipped). Is it make sense?
BTW, in the real project all of this can be done into the CI/CD configs and share the and will sent the notification to the devs/testers when all tests are done and result is ready.

Reply:
If I understood correctly, I think I have now implemented the solution properly. Please check and let me know.